src/apps/testapps/testH3CellAreaExhaustive.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2020 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file
+ * @brief tests H3 cell area functions using tests over a large number
+ *        of indexes.
+ *
+ *  usage: `testH3CellAreaExhaustive`
+ */
+
+#include <math.h>
+
+#include "h3Index.h"
+#include "test.h"
+#include "utility.h"
+
+/**
+ * Basic checks around the great circle distance between the centers of two
+ * neighboring cells. Tests positivity and commutativity.
+ *
+ * Tests the functions:
+ *     pointDistRads
+ *     pointDistKm
+ *     pointDistM
+ *
+ * @param  edge  H3 unidirectional edge denoting neighboring cells
+ */
+static void haversine_assert(H3Index edge) {
+    GeoCoord a, b;
+
+    H3_EXPORT(h3ToGeo)
+    (H3_EXPORT(getOriginH3IndexFromUnidirectionalEdge)(edge), &a);
+    H3_EXPORT(h3ToGeo)
+    (H3_EXPORT(getDestinationH3IndexFromUnidirectionalEdge)(edge), &b);
+
+    char pos[] = "distance between cell centers should be positive";
+    char comm[] = "pairwise cell distances should be commutative";
+
+    double ab, ba;
+
+    ab = H3_EXPORT(pointDistRads)(&a, &b);
+    ba = H3_EXPORT(pointDistRads)(&b, &a);
+    t_assert(ab > 0, pos);
+    t_assert(ab == ba, comm);
+
+    ab = H3_EXPORT(pointDistKm)(&a, &b);
+    ba = H3_EXPORT(pointDistKm)(&b, &a);
+    t_assert(ab > 0, pos);
+    t_assert(ab == ba, comm);
+
+    ab = H3_EXPORT(pointDistM)(&a, &b);
+    ba = H3_EXPORT(pointDistM)(&b, &a);
+    t_assert(ab > 0, pos);
+    t_assert(ab == ba, comm);
```

Comment:
Should be possible to assert that i.e. the same measurement in meters is numerically greater than the same in kilometers, etc.

Reply:
Added!